### DISCUSSION TITLE
边上右键打开菜单后如何知道 distance 值？

### DISCUSSION THREAD

**Discussion by esmadrider** [antvis/X6#3815] [Q&A] [answered]
https://x6.antv.vision/en/examples/edge/tool#context-menu

如示例可以在边上右键，打开一个菜单。但只知道 x, y 的画布值。如何知道 distance 值？

我希望知道这个 distance 值后，在此处手动创建一个标签。就像双击的创建的效果一样：https://x6.antv.vision/en/examples/node/tool#editable

**Comment by NewByVector** [comment, marked as answer]
可以使用以下 API 获取在连线上距离 （x、y）最近的一个点距离起点的 distance。
```ts
const view = graph.findViewByCell(edge) as EdgeView
const distance = view.getClosestPointLength({ x, y })
```

**Reply by esmadrider** [reply]
好呀 感谢感谢
